scripts/annotate.py

```diff
@@ -30,23 +30,65 @@ def annotate_cells(
     configs:
         path to configuration file
     """
+    # initiating Logger
+    log_path = Path(log_file).absolute()
+    logging.basicConfig(
+        filename=log_path,
+        encoding="utf-8",
+        level=logging.DEBUG,
+        format="%(asctime)s.%(msecs)03d - %(levelname)s - %(thread)d - %(message)s",
+        datefmt="%Y-%m-%d %H:%M:%S",
+    )
+    logging.info("Starting Annotation processes")
 
     # loading in annotate configs
+    logging.info(f"Loading Annotation configuration from: {config}")
+
     annotate_path_obj = Path(config)
+    if not annotate_path_obj.is_file():
+        e_msg = "Unable to find Annotation configuration file"
+        logging.error(e_msg)
+        raise FileNotFoundError(e_msg)
+
     annotate_config_path = annotate_path_obj.absolute()
     with open(annotate_config_path, "r") as yaml_contents:
         annotate_configs = yaml.safe_load(yaml_contents)["annotate_configs"]["params"]
+        logging.info(f"Annotation configuration loaded")
 
-    # loading in platmap
-    plate = Path(aggregated_data).name.split("_")[0]
+    # loading in plate map
+    logging.info(f"Loading plate data from: {aggregated_data}")
+
+    logging.info(f"loadding barcodes from: {barcodes_path}")
     barcode_platemap_df = pd.read_csv(barcodes_path)
+
+    logging.info("Searching plate map name")
+    plate = Path(aggregated_data).name.split("_")[0]
     platemap = barcode_platemap_df.query(
         "Assay_Plate_Barcode == @plate"
     ).Plate_Map_Name.values[0]
+
+    # checking if plate map is found
+    if platemap == "" or platemap is None:
+        e_msg = "Unable to find associated plate map"
+        logging.error(e_msg)
+        raise ValueError(e_msg)
+
+    logging.info(f"Plate map found: {platemap}")
+
+    # loading for plate file
+    logging.info("loading plate map file")
     platemap_file = os.path.join(metadata_dir, "platemap", "{}.csv".format(platemap))
+    if not Path(platemap_file).is_file():
+        e_msg = "Unable to locate plate map file"
```

Comment:
```suggestion
        e_msg = f"Unable to locate plate map file: {platemap_file}"
```

To be more explicit (and therefore a more helpful error)

Reply:
you can probably add these explicit error messages in other places as well (I didn't make suggestions everywhere it would be helpful)

Reply:
Yes this is possible. I will make note of your suggestion in #9 and work on it on the future.